airflow/utils/sqlalchemy.py

```diff
@@ -177,6 +179,14 @@ def process_result_value(self, value, dialect):
         return data
 
 
+def _is_dialect_good_for_row_locks(dialect: DefaultDialect) -> bool:
+    supports_for_update_of = dialect.name != "mysql" or dialect.supports_for_update_of
+    mariadb_10_6 = (
+        isinstance(dialect, MySQLDialect) and dialect._is_mariadb and dialect.server_version_info >= (10, 6)
+    )
```

Comment:
Emm not sure depending on `_is_mariadb` is a good idea.

Reply:
Unfortunately, it looks like there is no better way. 

In our version of sqlalchemy it realized this way --  https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_3_24/lib/sqlalchemy/dialects/mysql/base.py#L2686, so we can inline it, but this is most likely worse

In the modern version of sqlalchemy we can replace it with dialect.is_mariadb (and it is backward compatible)
https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/dialects/mysql/base.py#L2756